### ISSUE TITLE
BatchNorm2d after AdaptiveAvgPool2d

### ISSUE THREAD

**Issue by felixcrazzolara** [chenjun2hao/DDRNet.pytorch#29]
In the DAPPM module there is the following code for `self.scale4`:
```
nn.AdaptiveAvgPool2d((1, 1)),
nn.BatchNorm2d(inplanes, momentum=bn_mom),
```
I get an error saying `Expected more than 1 value per channel when training, got input size torch.Size([1, 512, 1, 1])`. I am not surprised since the `AdaptiveAvgPool2d` block reduces height and width of the layers to just 1 as can be seen in the error message as well. There is no way that the BatchNorm2d layer can then compute the standard deviation. How is this supposed to work?

**Comment by felixcrazzolara** [comment]
I had to do `model.eval()`, which makes sense.

**Comment by Daniu-jiao** [comment]
> I had to do `model.eval()`, which makes sense.



Because the batch size you entered is 1
